glean-core/src/metrics/experiment.rs

```diff
@@ -101,23 +117,38 @@ impl ExperimentMetric {
         }
 
         // Make sure that branch id is within the expected limit.
-        let truncated_branch = if branch.len() > MAX_EXPERIMENTS_IDS_LEN {
+        let truncated_branch = if branch.len() > MAX_EXPERIMENTS_BRANCH_LEN {
             log::warn!(
                 "Value length {} for branch exceeds maximum of {}",
                 branch.len(),
-                MAX_EXPERIMENTS_IDS_LEN
+                MAX_EXPERIMENTS_BRANCH_LEN
             );
-            truncate_string_at_boundary(branch, MAX_EXPERIMENTS_IDS_LEN)
+            truncate_string_at_boundary(branch, MAX_EXPERIMENTS_BRANCH_LEN)
         } else {
             branch
         };
 
-        // TODO (bug 1566001): add sane limits for the extra keys and
-        // values as well.
+        // Apply limits to extras
+        let truncated_extras = if extra.is_some() {
+            let mut temp_map = HashMap::new();
+            for (key, value) in extra.unwrap().into_iter() {
+                let truncated_key = truncate_string_at_boundary(key, MAX_EXPERIMENT_KEY_LEN);
+                let truncated_value = truncate_string_at_boundary(value, MAX_EXPERIMENT_VALUE_LEN);
+                temp_map.insert(truncated_key, truncated_value);
+            }
```

Comment:
You could avoid converting all values only to throw some of them away by integrating the `take` here as well.

```
extra.unwrap().into_iter().take(MAX_EXPERIMENTS_EXTRAS_LEN)
```

Reply:
That's less overhead, and then I shouldn't even need to apply `take` again if the list is already truncated to the correct number of entries.